src/clipboard.rs

```diff
@@ -0,0 +1,294 @@
+/// asked to copy to different destination sequences.
+///
+/// | Terminal (Version)  | Test System | Copy to ''  | Copy to 'c'  | Copy to 'p' | Copy to 'cp'       | Copy to 'pc'       |
+/// | ------------------- | ----------- | ----------- | ------------ | ----------- | ------------------ | ------------------ |
+/// | xterm (397)         | 1           | primary     | clipboard    | primary     | clipboard, primary | clipboard, primary |
+/// | Alacritty (0.15.1)  | 1           | clipboard   | clipboard    | primary     | clipboard          | primary            |
+/// | Wezterm (*1)        | 1           | clipboard   | clipboard    | primary     | clipboard          | clipboard          |
+/// | Konsole (24.12.3)   | 1           | clipboard   | clipboard    | primary     | clipboard, primary | clipboard, primary |
+/// | Kitty (0.40.0)      | 1           | clipboard   | clipboard    | primary     | clipboard          | clipboard          |
+/// | foot (1.20.2)       | 1           | clipboard   | clipboard    | primary     | clipboard, primary | clipboard, primary |
+/// | tmux (3.5a)         | 1, *2       | primary     | clipboard    | primary     | clipboard, primary | clipboard, primary |
+///
+/// Test systems were the following:
+/// 1. Wayland with [primary selection protocol](https://wayland.app/protocols/primary-selection-unstable-v1) enabled.
+///
+/// Asterisks:
+/// 1. 20240203-110809-5046fc22
+/// 2. set-clipboard set to external, i.e. this is OSC52 pass-through
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct CopyToClipboard<T> {
+    /// Content to be copied
+    pub content: T,
+    /// Sequence of copy destinations
+    ///
+    /// Not all sequences are equally supported by terminal emulators. See
+    /// [`CopyToClipboard` (Terminal Support)](struct.CopyToClipboard.html#terminal-support).
+    pub destination: ClipboardSelection,
+}
+
+impl<T: AsRef<[u8]>> Command for CopyToClipboard<T> {
+    fn write_ansi(&self, f: &mut impl fmt::Write) -> fmt::Result {
+        write!(
+            f,
+            osc!("52;{destination};{encoded_text}"),
+            destination = self.destination.to_string(),
+            encoded_text = BASE64_STANDARD.encode(&self.content)
+        )
+    }
+
+    #[cfg(windows)]
+    fn execute_winapi(&self) -> std::io::Result<()> {
+        use std::io;
+
+        Err(io::Error::new(
+            io::ErrorKind::Unsupported,
+            "Copying is not implemented for the Windows API.",
+        ))
+    }
+}
+
+impl<T: AsRef<[u8]>> CopyToClipboard<T> {
+    /// Construct a [`CopyToClipboard`] that writes content into the
+    /// "clipboard" (or 'c') clipboard selection.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use crossterm::{execute, Command};
+    /// use crossterm::clipboard::CopyToClipboard;
+    /// execute!(std::io::stdout(), CopyToClipboard::into_clipboard_from("foo"));
+    /// ```
+    pub fn into_clipboard_from(content: T) -> CopyToClipboard<T> {
```

Comment:
bikeshed: I think maybe to_clipboard / to_primary might be obvious enough names here. into_ gives greater connotations of conversion of the type into another type, which is not happening here. with_ would be the obvious builder style name, but that feels wrong here to me too.